src/lib/mailer/index.js

```diff
@@ -221,36 +276,279 @@ export const approverAssignedNotification = (report, newApprovers) => {
+        freq,
+        subjectFreq,
+      };
+
+      notificationDigestQueue.add(EMAIL_ACTIONS.SUBMITTED_DIGEST, data);
+      return data;
+    });
+    return Promise.all(records);
+  } catch (err) {
+    logger.info(`MAILER: submittedDigest with key ${USER_SETTINGS.EMAIL.KEYS.SUBMITTED_FOR_REVIEW} freq ${freq} error ${err}`);
+    throw err;
+  }
+}
+
+/**
+ * Finds users that are subscribed to the approved digest.
+ * For each user it retrieves the relevant report ids based on the timeframe.
+ *
+ * @param {String} freq - frequency of the approved digests (daily/weekly/monthly)
+ *
+ */
+export async function approvedDigest(freq, subjectFreq) {
+  let data = null;
+  const date = frequencyToInterval(freq);
+  logger.info(`MAILER: Starting ApprovedDigest with freq ${freq}`);
+  try {
+    if (!date) {
+      throw new Error('date is null');
+    }
+    // Find Users with preferences
+    const users = await usersWithSetting(USER_SETTINGS.EMAIL.KEYS.APPROVAL, [freq]);
+
+    const records = users.map(async (user) => {
+      const reports = await activityReportsApprovedByDate(user.id, date);
+
+      data = {
+        user,
+        reports,
+        type: EMAIL_ACTIONS.APPROVED_DIGEST,
+        freq,
+        subjectFreq,
+      };
+
+      notificationDigestQueue.add(EMAIL_ACTIONS.APPROVED_DIGEST, data);
+      return data;
+    });
+    return Promise.all(records);
+  } catch (err) {
+    logger.info(`MAILER: ApprovedDigest with key ${USER_SETTINGS.EMAIL.KEYS.APPROVAL} freq ${freq} error ${err}`);
+    throw err;
+  }
+}
+
+/**
+ * Retrieves the correct template based on parameters and send a digest email.
+ *
+ * @param {*} job - job containing data
+ * @param {*} transport - nodemailer transport
+ *
+ */
+export const notifyDigest = (job, transport = defaultTransport) => {
+  const {
+    user, reports, type, freq, subjectFreq,
+  } = job.data;
+
+  // Set these inside the function to allow easier testing
+  const { FROM_EMAIL_ADDRESS, SEND_NOTIFICATIONS } = process.env;
+
+  logger.debug(`MAILER: Creating ${user.email}'s ${type} digest for ${freq}`);
+
+  if (SEND_NOTIFICATIONS === 'true') {
+    const reportPath = `${process.env.TTA_SMART_HUB_URI}/activity-reports/`;
+
+    const templateType = reports && reports.length > 0 ? 'digest' : 'digest_empty';
+
+    const email = new Email({
+      message: {
+        from: FROM_EMAIL_ADDRESS,
+      },
+      send,
+      transport,
+      htmlToText: {
+        wordwrap: 120,
+      },
+    });
+    return email.send({
+      template: path.resolve(emailTemplatePath, templateType),
+      message: {
+        to: [user.email],
+      },
+      locals: {
+        user,
+        reports,
+        reportPath,
+        type,
+        freq,
+        subjectFreq,
+      },
+    });
+  }
+  return null;
+};
+
+/**
+ * @param {User} user
+ * @param {string} token
+ * @returns Promise<any>
+ */
+export const sendEmailVerificationRequestWithToken = (user, token) => {
+  const email = new Email({
```

Comment:
should this also be logged in the ```MailerLogs``` table?

Reply:
Added:
https://ocio-jira.acf.hhs.gov/browse/TTAHUB-1025